sql: flake in TestCreateStatsAfterSchemaChange

Found here on branch release-23.1: https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelExtendedCi/9534865?showRootCauses=false&expandBuildProblemsSection=true&expandBuildTestsSection=true

```
Failed
=== RUN   TestCreateStatsAfterSchemaChange
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/b287de7194e17a0386ee1775f0d198ba/logTestCreateStatsAfterSchemaChange3090999982
    test_log_scope.go:79: use -show-logs to present logs inline
    sql_runner.go:103:
          Error Trace:  github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:116
                              github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:311
                              github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/schema_changer_test.go:5478
          Error:        Received unexpected error:
                        query 'SELECT statistics_name, column_names, row_count, distinct_count, null_count
                        (1) attached stack trace
                          -- stack trace:
                          | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
                          |   github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:314
                          | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
                          |   github.com/cockroachdb/cockroach/pkg/testutils/soon.go:71
                          | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
                          |   github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:213
                          | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
                          |   github.com/cockroachdb/cockroach/pkg/testutils/soon.go:77
                          | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
                          |   github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:116
                          | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
                          |   github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:311
                          | github.com/cockroachdb/cockroach/pkg/sql_test.TestCreateStatsAfterSchemaChange
                          |   github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/schema_changer_test.go:5478
                          | testing.tRunner
                          |   GOROOT/src/testing/testing.go:1446
                          | runtime.goexit
                          |   src/runtime/asm_amd64.s:1594
                        Wraps: (2) query 'SELECT statistics_name, column_names, row_count, distinct_count, null_count
                          |     FROM [SHOW STATISTICS FOR TABLE t.test] ORDER BY column_names::STRING': expected:
                          | __auto__, {k}, 0, 0, 0
                          | __auto__, {v}, 0, 0, 0
                          | __auto__, {w}, 0, 0, 0
                          | got:
                          |
                        Error types: (1) *withstack.withStack (2) *errutil.leafError
    panic.go:540: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/b287de7194e17a0386ee1775f0d198ba/logTestCreateStatsAfterSchemaChange3090999982
--- FAIL: TestCreateStatsAfterSchemaChange (53.08s)
```



Jira issue: CRDB-26837

--- a/pkg/sql/schema_changer_test.go
+++ b/pkg/sql/schema_changer_test.go
@@ -5450,6 +5450,7 @@ CREATE TABLE t.test (a INT, b INT, c JSON, d JSON);
 func TestCreateStatsAfterSchemaChange(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
+	skip.UnderRace(t, "sometimes the timer in stats.Refresher doesn't fire fast enough under race")
 
 	defer func(oldRefreshInterval, oldAsOf time.Duration) {
 		stats.DefaultRefreshInterval = oldRefreshInterval

